cirq-core/cirq/ops/control_values.py

```diff
@@ -180,3 +180,80 @@ def __and__(self, other: AbstractControlValues) -> 'ProductOfSums':
                 f'And operation not supported between types ProductOfSums and {type(other)}'
             )
         return type(self)(self._internal_representation + other._internal_representation)
+
+
+@dataclass(frozen=True, eq=False)
+class SumOfProducts(AbstractControlValues):
+    """SumOfProducts represents control values in a form of a list or products.
+
+    SumOfProducts allows the creation of control values combinations that
+    can't be factored as the product of independent expressions (e.g. XOR)
+
+
+    Examples:
+        xor = SumOfProducts(((0, 1), (1, 0)))   # xor of two bits.
+        nand =  SumOfProducts(((0, 0), (0, 1), (1, 0))) # nand of two bits
+    """
```

Comment:
```suggestion

@dataclass(frozen=True, eq=False)
class SumOfProducts(AbstractControlValues):
    """Represents control values as a union of n-bit tuples, each describing an allowed assignment.

    `SumOfProducts` representation describes the control values as a union of n-bit tuples, where
    each n-bit tuple represents an allowed assignment of bits for which the control should be
    activated. This expanded representation allows us to create control values combinations which
    cannot be factored as a `ProductOfSums` representation.

    For example:

    1) `(|00><00| + |11><11|) X + (|01><01| + |10><10|) I` represents an operator which flips
        the third qubit if the first two qubits are `00` or `11`, and does nothing otherwise.
        This can be constructed as
            >>> xor_control_values = cirq.SumOfProducts(((0, 0), (1, 1)))
            >>> q0, q1, q2 = cirq.LineQubit.range(3)
            >>> cirq.X(q2).controlled_by(q0, q1, control_values=xor_control_values)

    2) `(|00><00| + |01><01| + |10><10|) X + (|11><11|) I` represents an operators which flips
        the third qubit if the `nand` of first two qubits is `1` (i.e. first two qubits are
        either `00`, `01` or `10`), and does nothing otherwise. This can be constructed as:
            >>> nand_control_values = cirq.SumOfProducts(((0, 0), (0, 1), (1, 0)))
            >>> q0, q1, q2 = cirq.LineQubit.range(3)
            >>> cirq.X(q2).controlled_by(q0, q1, control_values=nand_control_values)
    """
```

Reply:
done